client/my-sites/theme/main.jsx

```diff
@@ -635,11 +635,16 @@ class ThemeSheet extends Component {
 			showTryAndCustomize,
 			isExternallyManagedTheme,
 			isWporg,
+			isLoggedIn,
 		} = this.props;
 		const placeholder = <span className="theme__sheet-placeholder">loading.....</span>;
 		const title = name || placeholder;
 		const tag = author ? translate( 'by %(author)s', { args: { author: author } } ) : placeholder;
-		const shouldRenderButton = ! retired && ! isWPForTeamsSite && ! this.shouldRenderForStaging();
+		const shouldRenderButton =
+			! retired &&
+			! isWPForTeamsSite &&
+			! this.shouldRenderForStaging() &&
+			( ! this.hasWpOrgThemeUpsellBanner() || ! isLoggedIn );
```

Comment:
![badge](https://img.shields.io/badge/review-ask-blue.svg)

May I ask why `hasWpOrgThemeUpsellBanner` is needed also?

Reply:
That controls the presence of the "Access this third-party theme with the Business plan!" banner:

<img width="749" alt="Screenshot 2023-06-23 at 09 36 47" src="https://github.com/Automattic/wp-calypso/assets/1233880/62eb9450-6e0e-42ad-a74c-c8484a6908cd">

We want to hide the button when that banner is visible.

Reply:
Ahh, a logged-out user also has that banner, so we combined `( ! this.hasWpOrgThemeUpsellBanner() || ! isLoggedIn )`. Thank you for your comment!

My concern was described in [another comment](https://github.com/Automattic/wp-calypso/pull/78537#discussion_r1239460031), but non-clocking!